WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubTimeRangeSelection.swift

```diff
@@ -0,0 +1,138 @@
+        case .today:
+            self.selectionDelegate = AnalyticsHubDayRangeSelection(referenceDate: currentDate, currentCalendar: currentCalendar)
+        case .weekToDate:
+            self.selectionDelegate = AnalyticsHubWeekRangeSelection(referenceDate: currentDate, currentCalendar: currentCalendar)
+        case .monthToDate:
+            self.selectionDelegate = AnalyticsHubMonthRangeSelection(referenceDate: currentDate, currentCalendar: currentCalendar)
+        case .yearToDate:
+            self.selectionDelegate = AnalyticsHubYearRangeSelection(referenceDate: currentDate, currentCalendar: currentCalendar)
+        }
+    }
+
+    /// Unwrap the generated selected `AnalyticsHubTimeRange` based on the `selectedTimeRange`
+    /// provided during initialization.
+    /// - throws an `.selectedRangeGenerationFailed` error if the unwrap fails.
+    func unwrapCurrentTimeRange() throws -> AnalyticsHubTimeRange {
+        guard let currentTimeRange = selectionDelegate.currentTimeRange else {
+            throw TimeRangeGeneratorError.selectedRangeGenerationFailed
+        }
+        return currentTimeRange
+    }
+
+    /// Unwrap the generated previous `AnalyticsHubTimeRange`relative to the selected one
+    /// based on the `selectedTimeRange` provided during initialization.
+    /// - throws a `.previousRangeGenerationFailed` error if the unwrap fails.
+    func unwrapPreviousTimeRange() throws -> AnalyticsHubTimeRange {
+        guard let previousTimeRange = selectionDelegate.previousTimeRange else {
+            throw TimeRangeGeneratorError.previousRangeGenerationFailed
+        }
+        return previousTimeRange
+    }
+
+    /// Generates a date description of the previous time range set internally.
+    /// - Returns the Time range in a UI friendly format. If the previous time range is not available,
+    /// then returns an presentable error message.
+    func generateCurrentRangeDescription() -> String {
+        guard let currentTimeRangeDescription = selectionDelegate.currentRangeDescription else {
+            return Localization.noCurrentPeriodAvailable
+        }
+        return currentTimeRangeDescription
+    }
+
+    /// Generates a date description of the previous time range set internally.
+    /// - Returns the Time range in a UI friendly format. If the previous time range is not available,
+    /// then returns an presentable error message.
+    func generatePreviousRangeDescription() -> String {
+        guard let previousTimeRangeDescription = selectionDelegate.previousRangeDescription else {
+            return Localization.noPreviousPeriodAvailable
+        }
+        return previousTimeRangeDescription
+    }
+}
+
+// MARK: - Time Range Selection Type
+extension AnalyticsHubTimeRangeSelection {
+    enum SelectionType: CaseIterable {
+        case today
+        case weekToDate
+        case monthToDate
+        case yearToDate
+
+        var description: String {
+            get {
```

Comment:
nit: can be removed

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8267/commits/e456e6d45f2731ebd5fdfd459b1ecd11215a5a78